ext/AbstractDifferentiationEnzymeExt.jl

```diff
@@ -0,0 +1,62 @@
+module AbstractDifferentiationEnzymeExt
+
+if isdefined(Base, :get_extension)
+    import AbstractDifferentiation as AD
+    using Enzyme: Enzyme
+else
+    import ..AbstractDifferentiation as AD
+    using ..Enzyme: Enzyme
+end
+
+struct Mutating{F}
+    f::F
+end
+function (f::Mutating)(y, xs...)
+    y .= f.f(xs...)
+    return y
+end
+
+AD.@primitive function value_and_pullback_function(b::AD.EnzymeReverseBackend, f, xs...)
+    y = f(xs...)
```

Comment:
You should use ReverseSplitMode here, and call the augmented forward pass for that result, use the reverse pass (and tape created from aug) for the reverse pass.